src/MigrationLog/MigrationLogRepository.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace Give\MigrationLog;
+
+use Give\Framework\Database\DB;
+
+/**
+ * Class MigrationLogRepository
+ * @package Give\MigrationLog
+ *
+ * @since 2.10.0
+ */
+class MigrationLogRepository {
+
+	/**
+	 * @var string
+	 */
+	private $migration_table;
+
+	/**
+	 * @var MigrationLogFactory
+	 */
+	private $migrationFactory;
+
+	/**
+	 * MigrationRepository constructor.
+	 *
+	 * @param  MigrationLogFactory  $migrationFactory
+	 */
+	public function __construct( MigrationLogFactory $migrationFactory ) {
+		global $wpdb;
+		$this->migration_table  = "{$wpdb->prefix}give_migrations";
+		$this->migrationFactory = $migrationFactory;
+	}
+
+	/**
+	 * Save Migration
+	 *
+	 * @param  MigrationLogModel  $model
+	 */
+	public function save( MigrationLogModel $model ) {
+		$query = "
+			INSERT INTO {$this->migration_table} (id, status, error ) 
+			VALUES (%s, %s, %s) 
+			ON DUPLICATE KEY UPDATE
+			status = %s,
+			error = %s,
+			last_run = NOW()
+		";
+
+		DB::query(
+			DB::prepare(
+				$query,
+				$model->getId(),
+				$model->getStatus(),
+				$model->getError(),
+				$model->getStatus(),
+				$model->getError()
+			)
```

Comment:
@alaca Can we add placeholders in sql query? after that we do not have to call same function twice.

Reply:
I'm not sure what you're suggesting here, @ravinderk. He is using placeholders in the SQL query.

Reply:
@JasonTheAdams @alaca I suggest adding a number placeholder. This helps you to use the same value at multiple places in SQL queries.

```php
$query = "
			INSERT INTO {$this->migration_table} (id, status, error ) 
			VALUES (%1$s, %2$s, %3$s) 
			ON DUPLICATE KEY UPDATE
			status = %2$s,
			error = %3$s,
			last_run = NOW()
		";

DB::query(
	DB::prepare(
		$query,
		$model->getId(),
		$model->getStatus(),
		$model->getError()
		)
	)
}
			
```